pkg/apis/awsprovider/v1beta1/register.go

```diff
@@ -43,81 +43,66 @@ var (
 	SchemeBuilder = &scheme.Builder{GroupVersion: SchemeGroupVersion}
 )
 
-// AWSProviderConfigCodec is a runtime codec for the provider configuration
-// +k8s:deepcopy-gen=false
-type AWSProviderConfigCodec struct {
-	encoder runtime.Encoder
-	decoder runtime.Decoder
-}
-
-// NewScheme creates a new Scheme
-func NewScheme() (*runtime.Scheme, error) {
-	return SchemeBuilder.Build()
-}
-
-// NewCodec creates a serializer/deserializer for the provider configuration
-func NewCodec() (*AWSProviderConfigCodec, error) {
-	scheme, err := NewScheme()
-	if err != nil {
-		return nil, err
+// RawExtensionFromProviderSpec marshals the machine provider spec.
+func RawExtensionFromProviderSpec(spec *AWSMachineProviderConfig) (*runtime.RawExtension, error) {
+	if spec == nil {
+		return &runtime.RawExtension{}, nil
 	}
-	codecFactory := serializer.NewCodecFactory(scheme)
-	encoder, err := newEncoder(&codecFactory)
-	if err != nil {
-		return nil, err
-	}
-	codec := AWSProviderConfigCodec{
-		encoder: encoder,
-		decoder: codecFactory.UniversalDecoder(SchemeGroupVersion),
+
+	var rawBytes []byte
+	var err error
+	if rawBytes, err = json.Marshal(spec); err != nil {
+		return nil, fmt.Errorf("error marshalling providerSpec: %v", err)
 	}
-	return &codec, nil
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+	}, nil
 }
 
-// DecodeProviderSpec deserialises an object from the provider config
-func (codec *AWSProviderConfigCodec) DecodeProviderSpec(providerSpec *machinev1.ProviderSpec, out runtime.Object) error {
-	if providerSpec.Value != nil {
-		if _, _, err := codec.decoder.Decode(providerSpec.Value.Raw, nil, out); err != nil {
-			return fmt.Errorf("decoding failure: %v", err)
-		}
+// RawExtensionFromProviderStatus marshals the provider status
```

Comment:
For consistency with above
```suggestion
// RawExtensionFromProviderStatus marshals the machine provider status
```